src/FunderMaps.Data/RepositoryBase.cs

```diff
@@ -26,6 +27,112 @@ public virtual async Task<TEntity> AddGetAsync(TEntity entity)
             return await GetByIdAsync(primaryKey);
         }
 
+        /// <summary>
+        ///     Keypair used as cache identifier.
+        /// </summary>
+        protected class KeyPair
+        {
+            /// <summary>
+            ///     Entity hash key.
+            /// </summary>
+            public int EntityKey { get; set; }
+
+            /// <summary>
+            ///     Object hash key.
+            /// </summary>
+            public int Key { get; set; }
+
+            /// <summary>
+            ///     Keypair identifier.
+            /// </summary>
+            /// <remarks>
+            ///     This is very likely to overflow, however the identity shoud
+            ///     still be unique per keypair.
+            /// </remarks>
+            public int KeyPairIdentity => EntityKey + Key;
```

Comment:
Does this guarantee uniqueness? The hash code function doesn't guarantee the same length in its result and can be both positive and negative. I suggest modifying this in some way to quarantee uniqueness, for example something like

`$"{EntityKey}_{Key}"`

and making the key in our cache dictated by strings rather than by an integer.